### DISCUSSION TITLE
Redirects issue

### DISCUSSION THREAD

**Discussion by grakify90** [vercel/next.js#39805] [Help] [answered]
Hello!

I'm having trouble adding a redirect to my next.config file. I have a previously setup array of redirects that includes this one:
`  {
            "destination":  "/projecten",
            "permanent":  true,
            "source":  "/projects.php"
        }`

This works fine. However I needed to add more redirects, and this is where I can't get it figured out how to make it work correctly. When I add the following redirect, it will just pick up /projects.php and redirect to /projecten. However, I would want it to be picked up as /projects.php?beschikbaar=1 and be redirected to /woning-zoeken/beschikbare-woningen
`      {
            "destination":  "/woning-zoeken/beschikbare-woningen",
            "permanent":  true,
            "source":  "/projects.php\\?beschikbaar=1"
        },`
For clarification: I added the \\\ before the ? to escape it, otherwise nextJS would error right away.

Is there anyone who might be able to shed light on this? 

Thank you very much in advance!

**Comment by icyJoseph** [comment, marked as answer]
Hi,

I think you have to use the `has` property of redirects, like so:

```tsx
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  async redirects() {
    return [
      {
        source: "/projects.php",
        has: [
          {
            type: "query",
            key: "bar",
          },
        ],
        destination: "/buzz",
        permanent: false,
      },
      {
        source: "/projects.php",
        destination: "/foo",
        permanent: false,
      },
    ];
  },
};

module.exports = nextConfig;

```

Please refer to this section of the docs, https://nextjs.org/docs/api-reference/next.config.js/redirects#header-cookie-and-query-matching, for more information.

**Reply by grakify90** [reply]
Hi Joseph, thank you for your quick reply. I tried to implement it as you've explained:
![image](https://user-images.githubusercontent.com/63969844/185881523-f0af3ceb-6b28-478a-aa15-3bce187b400a.png)
But unfortunately /projects.php?beschikbaar=1 still redirects to /projecten (with the query string I provided, so actually /projecten?beschikbaar=1). Might I still be missing something?

**Reply by icyJoseph** [reply]
Hi,

Did you clear the cache? You had `permanent: true`, before, correct? The browser honours that... You need to reload the page with `cmd+shift+r`, or need to:

- open devTools, 
- and then a long press on the reload button, 
- and choose Hard Reload:

![Screenshot 2022-08-22 at 11 07 08](https://user-images.githubusercontent.com/21013447/185884021-9205476f-cb50-4c12-aa26-8a41c9fa47c9.png)

Does that fix it?

**Reply by grakify90** [reply]
Hi Joseph, that was indeed the issue! Now it's fixed :D Thank you so much for your help!
